Chap_API_Sync_Access.tex

```diff
@@ -1,171 +1,422 @@
+\end{itemize}
+
+Both of the above methods are based on retrieval from a specific process -
+i.e., the \refarg{proc} argument to \refapi{PMIx_Get} must include both the
+namespace and the rank of the process that posted the information. However, in
+some cases, non-reserved keys are provided on a globally unique basis and the
+retrieving process has no knowledge of the identity of the process posting the
+key. This is typically found in legacy applications (where the originating
+process identifier is often embedded in the key itself) and in unstructured
+applications that lack rank-related behavior. In these cases, the key remains
+associated with the namespace of the process that posted it, but is retrieved
+by use of the \refconst{PMIX_RANK_UNDEF} rank. In addition, the keys must be
+globally exchanged prior to retrieval as there is no way for the host to
+otherwise locate the source for the information.
+
+Note that the retrieval rules for non-reserved keys (detailed in Section 
+\ref{chap:data_sharing:retrules}) differ significantly from those used for reserved keys.
 
-This chapter covers both the synchronization and data retrieval functions
-provided under the \ac{PMIx} Standard.
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\section{\code{PMIx_Fence}}
-\declareapi{PMIx_Fence}
+\section{Posting Key/Value Pairs}
+\label{chap:data_sharing:set}
+
+\ac{PMIx} clients can post non-reserved key-value pairs associated with themselves by using \refapi{PMIx_Put}. Alternatively, \ac{PMIx} clients can cache both reserved and non-reserved key-value pairs accessible only by the caller via the \refapi{PMIx_Store_internal} \ac{API}.
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{\code{PMIx_Put}}
+\declareapi{PMIx_Put}
 
 %%%%
 \summary
 
-Execute a blocking barrier across the processes identified in the specified array, collecting information posted via \refapi{PMIx_Put} as directed.
+Stage a key/value pair in prepartion for being made accessible to processes.
 
 %%%%
 \format
 
-\copySignature{PMIx_Fence}{1.0}{
-pmix_status_t \\
-PMIx_Fence(const pmix_proc_t procs[], size_t nprocs, \\
-\hspace*{11\sigspace}const pmix_info_t info[], size_t ninfo);
-}
+\versionMarker{1.0}
+\cspecificstart
+\begin{codepar}
+pmix_status_t
+PMIx_Put(pmix_scope_t scope,
+         const pmix_key_t key,
+         pmix_value_t *val);
+\end{codepar}
+\cspecificend
```

Comment:
Editorial: I think we need to adjust this to use the `\copySignature` syntax used to support the Use Case appendix.

Reply:
There was no intention to change this, so I think this is a missed merge problem.